src/js/oncoprint.js

```diff
@@ -0,0 +1,125 @@
+var Track = require('./track').Track;
+var _ = require('underscore');
+var d3 = require('d3');
+var $ = require('jquery');
+module.exports = {};
+
+// TODO: use self everywhere
+
+var defaultOncoprintConfig = {
+	cell_width: 10,
+	cell_padding: 3,
+	render: 'table',
+};
+
+function Oncoprint(container_selector_string, config) {
+	var self = this;
```

Comment:
AFAIU there's nothing too tricky going on, and this is a pattern used by many.
When ```new Oncoprint()``` is called, javascript makes a new object and sets ```this``` to point to that object. So what I'm doing is essentially the same thing as what you're suggesting. By setting ```self``` we're getting around the context issues - ```self``` is wrapped into those functions by closure and refers specifically to that object - if we were to use the keyword ```this``` we'd have trickiness.

I'm actually not sure how you'd get around using ```this``` in your approach to refer to the object if you had a method on it that referred to another method on it.